pkg/kubelet/images/pullmanager/locks.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pullmanager
+
+import (
+	"hash/fnv"
+	"sync"
+)
+
+// StripedLockSet allows context locking based on string keys, where each key
+// is mapped to a an index in a size-limited slice of locks.
+type StripedLockSet struct {
+	locks []sync.Mutex
+	size  int32
+}
+
+// NewStripedLockSet creates a StripedLockSet with `size` number of locks to be
+// used for locking context based on string keys.
+// The size will be normalized to stay in the <1, 31> interval.
```

Comment:
min of 1 makes sense, why a max of 31?

Reply:
I followed the original comment which suggested max of 10... Should this be unbounded from top instead?

Reply:
not necessarily, I just wondered what the significance of 31 was and how it was chosen